gpytorch/lazy/lazy_evaluated_kernel_tensor.py

```diff
@@ -161,21 +161,21 @@ def _getitem(self, row_index, col_index, *batch_indices):
         # However - if we have multiple outputs per input, then the indices won't directly
         # correspond to the entries of row/col. We'll have to do a little pre-processing
         if num_outs_per_in_rows != 1 or num_outs_per_in_cols != 1:
-            if not isinstance(x1, slice) or not isinstance(x2, slice):
+            if not isinstance(row_index, slice) or not isinstance(col_index, slice):
                 # It's too complicated to deal with tensor indices in this case - we'll use the super method
                 return self.evaluate_kernel()._getitem(row_index, col_index, *batch_indices)
 
             # Now we know that x1 and x2 are slices
             # Let's make sure that the slice dimensions perfectly correspond with the number of
             # outputs per input that we have
             row_start, row_end, row_step = (
-                row_index.start,
-                row_index.stop,
+                row_index.start if row_index.start is not None else 0,
+                row_index.stop if row_index.stop is not None else self.shape[-2],
```

Comment:
I like the explicitness but this would also work (same below)
```suggestion
                row_index.start or 0,
                row_index.stop or self.shape[-2],
```